mix/qml/Debugger.qml

```diff
@@ -4,266 +4,329 @@ import QtQuick.Controls 1.1
+			errorLine.text = errorInfo.errorLine;
+		}
+		forceActiveFocus();
+	}
 
-		ListView {
+	Rectangle
+	{
+		visible: false;
+		id: compilationErrorArea
+		width: parent.width - 20
+		height: 500
+		color: "#ededed"
+		anchors.left: parent.left
+		anchors.top: parent.top
+		anchors.margins: 10
+		ColumnLayout
+		{
+			width: parent.width
 			anchors.top: parent.top
-			height: parent.height * 0.60
-			width: 200
-			anchors.horizontalCenter: parent.horizontalCenter
-			id: statesList
-			model: humanReadableExecutionCode
-			delegate: renderDelegate
-			highlight: highlightBar
-			highlightFollowsCurrentItem: true
-		}
+			spacing: 25
+			RowLayout
+			{
+				height: 100
+				Image {
+					id: compileFailed
+					source: "qrc:/qml/img/compilfailed.png"
+				}
+				ColumnLayout
+				{
+					Text {
+						color: "red"
+						id: errorLocation
+					}
+					Text {
+						color: "#4a4a4a"
+						id: errorDetail
+					}
+				}
+			}
 
-		Component {
-			id: highlightBar
-			Rectangle {
-				height: statesList.currentItem.height
-				width: statesList.currentItem.width
-				border.color: "orange"
-				border.width: 1
-				Behavior on y { SpringAnimation { spring: 2; damping: 0.1 } }
+			Rectangle
+			{
+				width: parent.width - 6
+				height: 2
+				color: "#d0d0d0"
 			}
-		}
 
-		Component {
-			id: renderDelegate
-			Item {
-				id: wrapperItem
-				height: 20
-				width: parent.width
-				Text {
-					anchors.centerIn: parent
-					text: line
-					font.pointSize: 9
+			RowLayout
+			{
+				Text
+				{
+					color: "#4a4a4a"
+					id: errorLine
 				}
 			}
 		}
+	}
 
-		Rectangle {
-			id: callStackPanel
-			anchors.top: statesList.bottom
-			height: parent.height * 0.35
-			width: parent.width
+
+	Flickable {
+		id: debugScrollArea
+		flickableDirection: Flickable.VerticalFlick
+		anchors.fill: parent
+		contentHeight: machineStates.height
+		contentWidth: machineStates.width
+
+		GridLayout
+		{
+			property int sideMargin: 10
+			id: machineStates
+			anchors.top: parent.top
 			anchors.topMargin: 15
-			color: "transparent"
-			Label {
-				id: callStackLabel
-				anchors.bottomMargin: 10
-				horizontalAlignment: "AlignHCenter"
-				font.family: "Verdana"
-				font.pointSize: 8
-				font.letterSpacing: 2
-				width: parent.width
-				height: 15
-				text: qsTr("callstack")
-			}
+			anchors.left: parent.left;
+			anchors.leftMargin: machineStates.sideMargin
+			anchors.right: parent.right;
+			anchors.rightMargin: machineStates.sideMargin
+			flow: GridLayout.TopToBottom
+			//columnSpacing: 7
+			rowSpacing: 15
+			RowLayout {
+				// step button + slider
+				spacing: 10
+				height: 27
+				width: debugPanel.width
+				RowLayout {
+					id: jumpButtons
+					spacing: 15
+					width: 250
+					height: parent.height
+
+					StepActionImage
+					{
+						id: jumpoutbackaction;
+						source: "qrc:/qml/img/jumpoutback.png"
+						disableStateImg: "qrc:/qml/img/jumpoutbackdisabled.png"
+						onClicked: Debugger.stepOutBack()
```

Comment:
Better change StepActionImage to be a Button with image and assign Action objects instead of onClicked. This way you'll also get keyboard shortcuts